app/controllers/users_controller.rb

```diff
@@ -1,2 +1,30 @@
 class UsersController < ApplicationController
+  before_action :validate_session
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def follow
+    @user = User.find(params[:id])
+    @user.followers << current_user
+    redirect_to @user
+  end
+
+  def unfollow
+    @user = User.find(params[:id])
+    @user.received_follows.find_by(follower_id: current_user.id).destroy
```

Comment:
@idprietop what if the **find_by** method returns a nil ?

Reply:
@idprietop I meant this line `User.find(params[:id])`

Reply:
In case it returns nil, an undefined method error appears for the nil class as it will try to access the received_follows method.

Reply:
@idprietop how to prevent that error? please code it :)

Reply:
def unfollow
    @user = User.find(params[:id])
    if @user.nil?
      redirect_to articles_path
    else
      @user.received_follows.find_by(follower_id: current_user.id).destroy
      redirect_to @user
    end
  end
  
  this is the only way I see possible to control it because I think the @user could never become nil in the unfollow action.

Reply:
@idprietop please update your code!